src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(
+  items:T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
```

Comment:
Refactor it by using the reduce method. You don't really need to create a variable with an empty array